runtime/memprof.c

```diff
@@ -765,6 +765,25 @@ void caml_memprof_track_alloc_shr(value block)
   check_action_pending();
 }
 
+void caml_memprof_track_custom(value block, mlsize_t bytes)
+{
+  uintnat n_samples;
+  value callstack = 0;
+  CAMLassert(Is_young(block) || Is_in_heap(block));
+
+  if (lambda == 0 || local->suspended) return;
+
+  n_samples = rand_binom(Wsize_bsize(bytes));
+  if (n_samples == 0) return;
+
+  callstack = capture_callstack_postponed();
+  if (callstack == 0) return;
+
+  new_tracked(n_samples, Wsize_bsize(bytes), 0, Is_young(block),
+              block, callstack);
+  check_action_pending();
+}
```

Comment:
There is now a fair amount of duplication between `caml_memprof_track_shr` and `caml_memprof_track_custom`, would it make sense to share code?

```c
void caml_memprof_track(size, is_unmarshalled, is_young, block)
{
  uintnat n_samples;
  value callstack = 0;

  if (lambda == 0 || local->suspended) return;

  n_samples = rand_binom(size);
  if (n_samples == 0) return;

  callstack = capture_callstack_postponed();
  if (callstack == 0) return;

  new_tracked(n_samples, size, is_unmarshalled, is_young, block, callstack);
  check_action_pending();
}
```

Reply:
Done. Turns out there was a third occurrence of more-or-less the same code as well.